core-web/libs/sdk/analytics/src/lib/dotAnalytics/shared/dot-content-analytics.model.ts

```diff
@@ -26,146 +26,319 @@ export interface DotContentAnalyticsConfig {
     autoPageView?: boolean;
 
     /**
-     * The API key for authenticating with the Analytics service.
+     * The site key for authenticating with the Analytics service.
      */
-    apiKey: string;
+    siteKey: string;
 
     /**
      * Custom redirect function handler.
      * When provided, this function will be called instead of the default browser redirect
      * for handling URL redirections.
-     *
-     * @param {string} url - The URL to redirect to
      */
     redirectFn?: (url: string) => void;
 }
 
-// UTM parameters generated from the expected UTM keys
-type UTMParams = {
-    [key in (typeof EXPECTED_UTM_KEYS)[number] as key extends `utm_${infer U}`
-        ? U
-        : never]?: string;
-};
+/**
+ * Individual analytics event structure for DotCMS.
+ * Represents a single event within an analytics request sent to the DotCMS analytics server.
+ */
+export interface DotCMSAnalyticsEvent {
+    /** The type of event being tracked */
+    event_type: 'pageview' | 'track';
+    /** Page data associated with the event */
+    page: DotCMSPageData;
```

Comment:
Isn't that this will be dependant if the `event_type === 'pageview'`?

Reply:
fixed.